### PR TITLE
W-11416423: Error declaration for extensions built with the Mule SDK

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

**Review by asanguinetti — CHANGES_REQUESTED** [review]
Partial review. Some minor things.

File: modules/extensions-mule-support/src/test/resources/extensions/extension-with-errors.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<extension xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xmlns="http://www.mulesoft.org/schema/mule/mule-extension"
+           xmlns:operation="http://www.mulesoft.org/schema/mule/operation"
+           xmlns:core="http://www.mulesoft.org/schema/mule/core"
```
[the remark this answers is not in the document]
Comment (eze210) [reply modules/extensions-mule-support/src/test/resources/extensions/extension-with-errors.xml:5]: I like it more with core

File: modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/ExtensionModelTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        .setArtifactCoordinates(TEST_ARTIFACT_COORDINATES)
+        .build();
+    return new MuleSdkPluginExtensionModelLoader().loadExtensionModel(loadingRequest);
+  }
+
+  /**
+   * Creates a JavaSDK plugin's {@link ExtensionModel} for the main class passed in the parameter {@code extensionFile}.
+   * 
+   * @param extensionClass main class of the extension.
+   * @param dependencies   classLoader to be used to load the plugin.
```
[the remark this answers is not in the document]
Comment (eze210) [reply modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/ExtensionModelTestUtils.java:148]: Done

File: modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/ExtensionModelTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+   * Creates a MuleSDK plugin's {@link ExtensionModel} for the file passed in the parameter {@code extensionFile}.
+   * 
+   * @param extensionFile            name of the file declaring an extension with the MuleSDK.
+   * @param classLoader              classLoader to be used to load the plugin.
+   * @param astParserExtensionModels set of extension models available for the {@link DslResolvingContext}.
+   * @return the corresponding {@link ExtensionModel}.
+   */
+  public static ExtensionModel loadMuleSdkExtension(String extensionFile,
+                                                    ClassLoader classLoader,
+                                                    Set<ExtensionModel> astParserExtensionModels) {
```
[the remark this answers is not in the document]
Comment (eze210) [reply modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/ExtensionModelTestUtils.java:134]: Added both versions of the method. We had a test that needed to specify the classloader

**Review by asanguinetti — APPROVED** [review]

**Review by asanguinetti — APPROVED** [review]

**Comment by eze210** [comment]
--validate

**Pull request merged**